src/tree/hashes.rs

```diff
@@ -237,8 +237,8 @@ impl RatchetTree {
         }
         let right = child;
 
-        // "If R is a leaf node, the check fails"
-        if right.is_leaf() {
+        // "If R is a blank leaf node, the check fails"
+        if self.nodes[child].is_blank() && right.is_leaf() {
```

Comment:
Not that it makes a difference for now, but I'd suggest to use `right` as an index for `self.nodes` for clarity.

Reply:
I'll fix it when I create tests for this issue. What do you mean by "real proof of the algorithm"? I fixed this, because I ran into cases that wouldn't work without this fix.